--- mDNSResponder-98/mDNSCore/mDNSEmbeddedAPI.h.orig	2005-01-27 23:57:55.000000000 +0100
+++ mDNSResponder-98/mDNSCore/mDNSEmbeddedAPI.h	2005-04-23 19:20:56.000000000 +0200
@@ -1069,7 +1069,7 @@
 //   Macro Name __LP64__ Value 1
 // A quick Google search for "defined(__LP64__)" OR "#ifdef __LP64__" gives 2590 hits and
 // a search for "#if __LP64__" gives only 12, so I think we'll go with the majority and use defined()
-#if defined(_LP64) || defined(__LP64__)
+#if defined(__alpha__) || defined(_LP64) || defined(__LP64__)
 typedef   signed int   mDNSs32;
 typedef unsigned int   mDNSu32;
 #elif defined(_ILP64) || defined(__ILP64__)
