--- mDNSResponder-98/mDNSPosix/Makefile.orig	2005-03-26 18:34:02.017295000 +0100
+++ mDNSResponder-98/mDNSPosix/Makefile	2005-03-26 18:49:31.538987072 +0100
@@ -254,6 +254,9 @@
 JAVACFLAGS_OS += -I$(JDK)/include/linux
 OPTIONALTARG = nss_mdns
 OPTINSTALL   = InstalledNSS
+LD = $(CC) -shared
+LIBSONAME = -Wl,-soname=libdns_sd.$(LDSUFFIX).$(LIBVERS)
+NSSSONAME = -Wl,-soname=$(NSSLINKNAME)
 else
 
 ifeq ($(os),netbsd)
@@ -404,16 +407,16 @@
 CLIENTLIBOBJS = $(OBJDIR)/dnssd_clientlib.c.so.o $(OBJDIR)/dnssd_clientstub.c.so.o $(OBJDIR)/dnssd_ipc.c.so.o
 
 $(BUILDDIR)/libdns_sd.$(LDSUFFIX): $(CLIENTLIBOBJS)
-	@$(LD) $(LIBFLAGS) -o $@ $+
-	@$(STRIP) $@
+	$(LD) $(LIBSONAME) $(LIBFLAGS) -o $@ $+
+	$(STRIP) $@
 
 # nss_mdns target builds the Name Service Switch module
 nss_mdns: setup $(BUILDDIR)/$(NSSLIBFILE)
 	@echo "Name Service Switch module done"
 
 $(BUILDDIR)/$(NSSLIBFILE): $(CLIENTLIBOBJS) $(OBJDIR)/nss_mdns.c.so.o
-	@$(LD) $(LIBFLAGS) -o $@ $+
-	@$(STRIP) $@
+	$(LD) $(NSSSONAME) $(LIBFLAGS) -o $@ $+
+	$(STRIP) $@
 
 #############################################################################
 
@@ -465,11 +468,9 @@
 
 $(NSSINSTPATH)/$(NSSLINKNAME): $(NSSINSTPATH)/$(NSSLIBFILE)
 	$(LN) $< $@
-	ldconfig
 
 $(NSSINSTPATH)/$(NSSLIBFILE): $(BUILDDIR)/$(NSSLIBFILE)
 	$(CP) $< $@
-	chmod 444 $@
 
 /etc/nss_mdns.conf: nss_mdns.conf
 	$(CP) $< $@
