--- mDNSResponder-576.30.4/mDNSPosix/Makefile.orig	2015-11-29 17:00:33.618091078 +0100
+++ mDNSResponder-576.30.4/mDNSPosix/Makefile	2015-11-29 17:10:11.764733480 +0100
@@ -106,6 +106,8 @@
 LD = $(CC) -shared
 FLEXFLAGS_OS = -l
 JAVACFLAGS_OS += -I$(JDK)/include/linux
+LIBSONAME = -Wl,-soname=libdns_sd.$(LDSUFFIX).$(LIBVERS)
+NSSSONAME = -Wl,-soname=$(NSSLINKNAME)
 
 # uClibc does not support Name Service Switch
 ifneq ($(os),linux-uclibc)
@@ -257,8 +259,8 @@
 CLIENTLIBOBJS = $(OBJDIR)/dnssd_clientlib.c.so.o $(OBJDIR)/dnssd_clientstub.c.so.o $(OBJDIR)/dnssd_ipc.c.so.o
 
 $(BUILDDIR)/libdns_sd.$(LDSUFFIX): $(CLIENTLIBOBJS)
-	@$(LD) $(LINKOPTS) -o $@ $+
-	@$(STRIP) $@
+	$(LD) $(LIBSONAME) $(LINKOPTS) -o $@ $+
+	$(STRIP) $@
 
 Clients: setup libdns_sd ../Clients/build/dns-sd
 	@echo "Clients done"
@@ -271,8 +273,8 @@
 	@echo "Name Service Switch module done"
 
 $(BUILDDIR)/$(NSSLIBFILE): $(CLIENTLIBOBJS) $(OBJDIR)/nss_mdns.c.so.o
-	@$(LD) $(LINKOPTS) -o $@ $+
-	@$(STRIP) $@
+	$(LD) $(NSSSONAME) $(LINKOPTS) -o $@ $+
+	$(STRIP) $@
 
 #############################################################################
 
@@ -347,11 +349,9 @@
 
 $(NSSINSTPATH)/$(NSSLINKNAME): $(NSSINSTPATH)/$(NSSLIBFILE)
 	$(LN) $< $@
-	ldconfig
 
 $(NSSINSTPATH)/$(NSSLIBFILE): $(BUILDDIR)/$(NSSLIBFILE)
 	$(CP) $< $@
-	chmod 444 $@
 
 /etc/nss_mdns.conf: nss_mdns.conf
 	$(CP) $< $@
